File: java/02_cryptographic_failures.java
// A02 - Cryptographic Failures. VULN: DES + static IV.
import javax.crypto.*;
import javax.crypto.spec.*;

public class CryptographicFailures {
  private static final byte[] KEY = "12345678".getBytes(); // VULN: weak key

  public static byte[] encrypt(String plain) throws Exception {
    Cipher c = Cipher.getInstance("DES/ECB/PKCS5Padding"); // VULN: DES + ECB
    c.init(Cipher.ENCRYPT_MODE, new SecretKeySpec(KEY, "DES"));
    return c.doFinal(plain.getBytes());
  }
}

File: php/02_cryptographic_failures.php
<?php
// A02 - Cryptographic Failures. VULN: ECB AES + key in source.
// VULN: A02
$key = "0123456789abcdef";
function encrypt_token($plain) {
    global $key;
    return openssl_encrypt($plain, "AES-128-ECB", $key);
}
